component/loki/source/docker/runner.go

```diff
@@ -0,0 +1,236 @@
+package docker
+
+import (
+	"context"
+	"sync"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/agent/component/common/loki"
+	"github.com/grafana/agent/component/common/loki/positions"
+	dt "github.com/grafana/agent/component/loki/source/docker/internal/dockertarget"
+	"github.com/grafana/agent/pkg/runner"
+	"github.com/prometheus/common/model"
+)
+
+// A Manager manages a set of running tailers.
+type Manager struct {
+	log log.Logger
+
+	mut   sync.Mutex
+	opts  *Options
+	tasks []*tailerTask
+
+	runner *runner.Runner[*tailerTask]
+}
+
+// NewManager returns a new Manager which manages a set of running tailers.
+// Options must not be modified after passing it to a Manager.
+//
+// If NewManager is called with a nil set of options, no targets will be
+// scheduled for running until UpdateOptions is called.
+func NewManager(l log.Logger, opts *Options) *Manager {
```

Comment:
opts is unused here, is that intentional?

Reply:
Whoops didn't notice this, I'd just copy/pasted from the `kubetail` package; I'm fixing this on both instances.